routes/checkAuth.js

```diff
@@ -0,0 +1,32 @@
+let jwt = require('jsonwebtoken');
+
+let express = require('express');
+let router = express.Router();
+
+const validateToken = (req, res, next) => {
+  let authHeader = req.headers.authorization;
+  if (!authHeader) {
+    return res.sendStatus(401)
+  }
+  let tokenString = authHeader.split(" ");
+  if (tokenString[0] !== "Bearer"){
+    return res.sendStatus(401)
+  }
+
+  let token = tokenString[1];
+  jwt.verify(token, 'MY_SECRET', function(err, decoded) {
+    if (err){
+      return res.sendStatus(401)
+    }
+    res.locals.username = decoded.username;
```

Comment:
could you tell me the reason why you attach the username to response instead of request?

Reply:
Technically, it looks like "req.locals.username" make more sense, but "Express" only has res.locals, no req.locals.
http://expressjs.com/en/api.html#res.locals